### PR TITLE
Added the append string feature for ragg2

### PR DESCRIPTION
**Pull request by deepakchethan**
For feature #12862 
Please let me know if any changes are to be made.

If it's fine I'll update the radare2book as well.

### CODE REVIEW

File: binr/ragg2/ragg2.c [hunk 1]
```diff
@@ -40,6 +40,7 @@ static int usage(int v) {
 			" -q [fragment]   debruijn pattern offset\n"
 			" -r              show raw bytes instead of hexpairs\n"
 			" -s              show assembler\n"
+            		" -S              append a string\n"
```
Comment (radare) [inline binr/ragg2/ragg2.c]: That help is wrong. Doesnt implies the flag have any arg

File: binr/ragg2/ragg2.c [hunk 2]
```diff
@@ -226,6 +228,9 @@ int main(int argc, char **argv) {
 			}
 			}
 			break;
+		case 'S':
+		    str = optarg;
```
Comment (radare) [inline binr/ragg2/ragg2.c]: Use tabs

File: binr/ragg2/ragg2.c [hunk 3]
```diff
@@ -400,6 +405,15 @@ int main(int argc, char **argv) {
 		}
 	}
 
+	// append the provided string
+	if (str) {
+	    int l;
+	    l = strlen (str);
```
Comment (radare) [inline binr/ragg2/ragg2.c]: Define and assign in the same line and use tabs to indent

File: binr/ragg2/ragg2.c (hunk 1 above)
Comment (deepakchethan) [reply binr/ragg2/ragg2.c]: Okay, I'll add the [string] next to it.

File: binr/ragg2/ragg2.c (hunk 2 above)
Comment (deepakchethan) [reply binr/ragg2/ragg2.c]: I did use tabs, maybe it's the vscode thing?

File: binr/ragg2/ragg2.c (hunk 3 above)
Comment (deepakchethan) [reply binr/ragg2/ragg2.c]: Okay, I'll define and assign in the same line.

**Comment by radare** [comment]
Dont let your text editor lie to you. This is not a tab

File: binr/ragg2/ragg2.c [hunk 4]
```diff
@@ -400,6 +405,14 @@ int main(int argc, char **argv) {
 		}
 	}
 
+	// append the provided string
+	if (str) {
+		int l = strlen(str);
```
Comment (radare) [inline binr/ragg2/ragg2.c]: missing space before (

Comment (deepakchethan) [reply binr/ragg2/ragg2.c]: Okay my bad, I'll fix it.
Thank you for your review :)

**Pull request merged**